pipeline-modules/deployment-service/gcp/istio-gke-regional.yaml

```diff
@@ -474,9 +479,37 @@ template: |
     - '-c'
     - |
       cd {{helm_chart_path}} ;
-      helm upgrade --install -f values.yaml {{service_name}} . --atomic --timeout {{deployment_timeout}}
+      helm upgrade --install -f values.yaml {{service_name}} . --wait --timeout {{deployment_timeout}}
+    {% if rollback_enable %}
+      #Get Status code
+      helm_upgrade_exit_status=$?
+      echo 'Helm release failed with EXIT_CODE: '$helm_upgrade_exit_status''
+      # Check the Helm upgrade/install status
+      if [ $helm_upgrade_exit_status -eq 0 ]; then
+        echo "Helm upgrade/install successful...."
+      else
+        #If helm release failed than rollback
+        echo "Helm upgrade/install failed. Rolling back pervious version..."
+        
+        # Perform the Helm rollback
+        helm  rollback {{service_name}}  0 --wait --timeout {{deployment_timeout}}
```

Comment:
Why are we rolling back to version 0?

Reply:
If the second argument is omitted or set to 0, it will roll back to the previous release.

Reply:
`124             Wed Oct 25 17:41:23 2023        superseded      charts-0.1.0    1.16.0          Upgrade complete

125             Thu Oct 26 06:59:20 2023        failed          charts-0.1.0    1.16.0          Upgrade "evidence-service-stance-dev" failed: timed out wai
ting for the condition
126             Thu Oct 26 07:04:32 2023        superseded      charts-0.1.0    1.16.0          Rollback to 124`